FP-Erlang/Week 6/Assignment 5/Ashwin/assgn5.erl

```diff
@@ -0,0 +1,122 @@
+%########################################### ASSIGNMENT 5 ###########################################
+%########################################### Ashwin Joisa ###########################################
+-module(assgn5).
+
+% ############################################################################# Question 1
+-export([add/1, subtract/1, multiply/1, divide/1]).
+-export([greaterThanX/1, lessThanX/1]).
+% Function 1
+-export([takewhilePred/1, twp/2]).
+% Function 2
+-export([dropwhilePred/1, dw/2]).
+% Function 3
+-export([map/2, mapF/1]).
+% Function 4
+-export([any/2, anyPred/1]).
+% Function 5
+-export([all/2, allPred/1]).
+% ############################################################################# Question 2
+-export([derive/2]).
+% ############################################################################# Question 3
+-export([integral/4, integral/5]).
+% ############################################################################# Question 4
+-export([foldl/3]).
+
+
+% ########################################################################################## Question 1
+
+% Part 1
+
+add(A) -> fun(X) -> X+A end.
+subtract(A) -> fun(X) -> A-X end.
+multiply(A) -> fun(X) -> X*A end.
+divide(A) -> fun(X) -> A/X end.
+
```

Comment:
Include two argument versions of all the functions by making calls to the curried functions